### DISCUSSION TITLE
How to keep state in the "books" example

### DISCUSSION THREAD

**Discussion by amitrotner** [csells/go_router#367] [Q&A]
Hi, 

I am trying to add a state keeping ability to the "books" example. 
I have followed https://gorouter.dev/nested-navigation#keeping-state and changed `author_list.dart` so the authors list will be 50 times longer (just for the demonstration) and added `AutomaticKeepAliveClientMixin`.

```dart
// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import 'package:flutter/material.dart';

import '../data.dart';

class AuthorList extends StatefulWidget {
  const AuthorList({
    required this.authors,
    this.onTap,
    Key? key,
  }) : super(key: key);

  final List<Author> authors;
  final ValueChanged<Author>? onTap;

  @override
  State<AuthorList> createState() => _AuthorListState();
}

class _AuthorListState extends State<AuthorList> with AutomaticKeepAliveClientMixin {
  @override
  Widget build(BuildContext context) {
    super.build(context);
    return ListView.builder(
      itemCount: widget.authors.length * 50,
      itemBuilder: (context, index) => ListTile(
        title: Text(
          widget.authors[index % widget.authors.length].name,
        ),
        subtitle: Text(
          '${widget.authors[index % widget.authors.length].books.length} books',
        ),
        onTap: widget.onTap != null ? () => widget.onTap!(widget.authors[index % widget.authors.length]) : null,
      ),
    );
  }

  @override
  bool get wantKeepAlive => true;
}
```

However, when I scroll down the authors page, move to another page and return to the authors page, the state is not preserved.

How can I modify this example to enable state preserving?

Thanks

**Comment by csells** [comment]
Interesting. That sample worked when @rydmike built it. @rydmike thoughts about why it might not be working?

**Reply by rydmike** [reply]
Looks correct above, so unless something changed in the Flutter SDK it seems to be one of those "this should work" things. Do you have sample repo that reproduces the problem? Then I can take a look at it. 

I will also verify that the GoRouter example still works, but perhaps you did that already?
